File: GymApp/composeApp/src/commonMain/kotlin/com/gym/gymapp/ui/components/MemberItem.kt
package com.gym.gymapp.ui.components

import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.Chat
import androidx.compose.material.icons.filled.Call
import androidx.compose.material.icons.filled.CalendarToday
import androidx.compose.material.icons.filled.Event
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.gym.gymapp.getPlatform

@Composable
fun MemberItem(
    name: String, 
    status: String, 
    isActuallyExpired: Boolean,
    joinDate: String, 
    expiryDate: String, 
    phone: String?, 
    imageUrl: String?,
    onClick: () -> Unit
) {
    Card(
        modifier = Modifier.fillMaxWidth().clickable { onClick() },
        shape = RoundedCornerShape(20.dp),
        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface)
    ) {
        Row(
            modifier = Modifier.padding(12.dp),
            verticalAlignment = Alignment.Top
        ) {
            ProfileImage(
                imageUrl = imageUrl,
                name = name,
                size = 44
            )

            Spacer(modifier = Modifier.width(12.dp))

            Column(modifier = Modifier.weight(1f)) {
                val displayStatus = if (isActuallyExpired) "EXPIRED" else status

                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Text(name, fontWeight = FontWeight.Bold, fontSize = 14.sp, color = MaterialTheme.colorScheme.onSurface)
                    
                    Surface(
                        color = if (isActuallyExpired) Color.Red.copy(alpha = 0.1f) 
                               else if (status == "ACTIVE") MaterialTheme.colorScheme.primary.copy(alpha = 0.15f) 
                               else Color.Red.copy(alpha = 0.1f),
                        shape = RoundedCornerShape(10.dp)
                    ) {
                        Text(
                            displayStatus,
                            modifier = Modifier.padding(horizontal = 8.dp, vertical = 4.dp),
                            color = if (isActuallyExpired || status != "ACTIVE") Color.Red else MaterialTheme.colorScheme.primary,
                            fontSize = 9.sp,
                            fontWeight = FontWeight.Black
                        )
                    }
                }
                
                Row(
                    modifier = Modifier.padding(top = 4.dp),
                    horizontalArrangement = Arrangement.spacedBy(16.dp),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    // Joined Date
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Icon(
                            Icons.Default.CalendarToday,
                            contentDescription = null,
                            modifier = Modifier.size(10.dp),
                            tint = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.5f)
                        )
                        Text(
                            " Joined: $joinDate",
                            fontSize = 10.sp,
                            color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.5f)
                        )
                    }

                    // Expiry Date
                    val expiryColor = when {
                        isActuallyExpired -> Color.Red
                        else -> MaterialTheme.colorScheme.primary.copy(alpha = 0.7f)
                    }

                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Icon(
                            Icons.Default.Event,
                            contentDescription = null,
                            modifier = Modifier.size(10.dp),
                            tint = expiryColor
                        )
                        Text(
                            " Expires: $expiryDate",
                            fontSize = 10.sp,
                            color = expiryColor,
                            fontWeight = FontWeight.Bold
                        )
                    }
                }

                if (!phone.isNullOrBlank()) {
                    Row(
                        modifier = Modifier.padding(top = 12.dp),
                        horizontalArrangement = Arrangement.spacedBy(8.dp)
                    ) {
                        // WhatsApp Button
                        Button(
                            onClick = {
                                val number = phone.replace("+", "").replace(" ", "")
                                getPlatform().openUrl("https://wa.me/$number")
                            },
                            modifier = Modifier.height(30.dp),
                            contentPadding = PaddingValues(horizontal = 12.dp, vertical = 0.dp),
                            colors = ButtonDefaults.buttonColors(
                                containerColor = Color(0xFF059669).copy(alpha = 0.12f), // Emerald Green
                                contentColor = Color(0xFF059669)
                            ),
                            shape = RoundedCornerShape(8.dp),
                            elevation = null
                        ) {
                            Icon(Icons.AutoMirrored.Filled.Chat, contentDescription = null, modifier = Modifier.size(14.dp))
                            Spacer(Modifier.width(6.dp))
                            Text("WhatsApp", fontSize = 10.sp, fontWeight = FontWeight.Bold)
                        }

                        // Call Button
                        Button(
                            onClick = {
                                getPlatform().openUrl("tel:$phone")
                            },
                            modifier = Modifier.height(30.dp),
                            contentPadding = PaddingValues(horizontal = 12.dp, vertical = 0.dp),
                            colors = ButtonDefaults.buttonColors(
                                containerColor = Color(0xFF4F46E5).copy(alpha = 0.12f), // Indigo
                                contentColor = Color(0xFF4F46E5)
                            ),
                            shape = RoundedCornerShape(8.dp),
                            elevation = null
                        ) {
                            Icon(Icons.Default.Call, contentDescription = null, modifier = Modifier.size(14.dp))
                            Spacer(Modifier.width(6.dp))
                            Text("Call", fontSize = 10.sp, fontWeight = FontWeight.Bold)
                        }
                    }
                }
            }
        }
    }
}

File: gym_flutter_app/lib/ui/widgets/member_item.dart
import 'package:flutter/material.dart';
import 'package:url_launcher/url_launcher.dart';
import 'profile_image.dart';

class MemberItem extends StatelessWidget {
  final String name;
  final String status;
  final bool isActuallyExpired;
  final String joinDate;
  final String expiryDate;
  final String? phone;
  final String? imageUrl;
  final VoidCallback onClick;

  const MemberItem({
    super.key,
    required this.name,
    required this.status,
    required this.isActuallyExpired,
    required this.joinDate,
    required this.expiryDate,
    this.phone,
    this.imageUrl,
    required this.onClick,
  });

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);
    final displayStatus = isActuallyExpired ? "EXPIRED" : status;

    return Card(
      margin: EdgeInsets.zero,
      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(20)),
      color: theme.colorScheme.surface,
      child: InkWell(
        onTap: onClick,
        borderRadius: BorderRadius.circular(20),
        child: Padding(
          padding: const EdgeInsets.all(12.0),
          child: Row(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              ProfileImage(
                imageUrl: imageUrl,
                name: name,
                size: 44,
              ),
              const SizedBox(width: 12),
              Expanded(
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Row(
                      mainAxisAlignment: MainAxisAlignment.spaceBetween,
                      children: [
                        Text(
                          name,
                          style: theme.textTheme.bodyLarge?.copyWith(
                            fontWeight: FontWeight.bold,
                            fontSize: 14,
                            color: theme.colorScheme.onSurface,
                          ),
                        ),
                        Container(
                          padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 4),
                          decoration: BoxDecoration(
                            color: isActuallyExpired
                                ? Colors.red.withValues(alpha: 0.1)
                                : (status == "ACTIVE"
                                    ? theme.colorScheme.primary.withValues(alpha: 0.15)
                                    : Colors.red.withValues(alpha: 0.1)),
                            borderRadius: BorderRadius.circular(10),
                          ),
                          child: Text(
                            displayStatus,
                            style: TextStyle(
                              color: isActuallyExpired || status != "ACTIVE"
                                  ? Colors.red
                                  : theme.colorScheme.primary,
                              fontSize: 9,
                              fontWeight: FontWeight.w900,
                            ),
                          ),
                        ),
                      ],
                    ),
                    const SizedBox(height: 4),
                    Row(
                      children: [
                        Icon(
                          Icons.calendar_today,
                          size: 10,
                          color: theme.colorScheme.onSurface.withValues(alpha: 0.5),
                        ),
                        Text(
                          " Joined: $joinDate",
                          style: TextStyle(
                            fontSize: 10,
                            color: theme.colorScheme.onSurface.withValues(alpha: 0.5),
                          ),
                        ),
                        const SizedBox(width: 16),
                        _ExpiryInfo(
                          isActuallyExpired: isActuallyExpired,
                          expiryDate: expiryDate,
                          primaryColor: theme.colorScheme.primary,
                        ),
                      ],
                    ),
                    if (phone != null && phone!.isNotEmpty) ...[
                      const SizedBox(height: 12),
                      Row(
                        children: [
                          _ActionButton(
                            label: "WhatsApp",
                            icon: Icons.chat_rounded,
                            color: const Color(0xFF059669),
                            onPressed: () {
                              final number = phone!.replaceAll("+", "").replaceAll(" ", "");
                              _launchUrl("https://wa.me/$number");
                            },
                          ),
                          const SizedBox(width: 8),
                          _ActionButton(
                            label: "Call",
                            icon: Icons.call,
                            color: const Color(0xFF4F46E5),
                            onPressed: () {
                              _launchUrl("tel:$phone");
                            },
                          ),
                        ],
                      ),
                    ],
                  ],
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }

  Future<void> _launchUrl(String url) async {
    final uri = Uri.parse(url);
    if (await canLaunchUrl(uri)) {
      await launchUrl(uri);
    }
  }
}

class _ExpiryInfo extends StatelessWidget {
  final bool isActuallyExpired;
  final String expiryDate;
  final Color primaryColor;

  const _ExpiryInfo({
    required this.isActuallyExpired,
    required this.expiryDate,
    required this.primaryColor,
  });

  @override
  Widget build(BuildContext context) {
    final expiryColor = isActuallyExpired ? Colors.red : primaryColor.withValues(alpha: 0.7);

    return Row(
      children: [
        Icon(
          Icons.event,
          size: 10,
          color: expiryColor,
        ),
        Text(
          " Expires: $expiryDate",
          style: TextStyle(
            fontSize: 10,
            color: expiryColor,
            fontWeight: FontWeight.bold,
          ),
        ),
      ],
    );
  }
}

class _ActionButton extends StatelessWidget {
  final String label;
  final IconData icon;
  final Color color;
  final VoidCallback onPressed;

  const _ActionButton({
    required this.label,
    required this.icon,
    required this.color,
    required this.onPressed,
  });

  @override
  Widget build(BuildContext context) {
    return SizedBox(
      height: 30,
      child: TextButton.icon(
        onPressed: onPressed,
        icon: Icon(icon, size: 14),
        label: Text(
          label,
          style: const TextStyle(fontSize: 10, fontWeight: FontWeight.bold),
        ),
        style: TextButton.styleFrom(
          backgroundColor: color.withValues(alpha: 0.12),
          foregroundColor: color,
          shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(8)),
          padding: const EdgeInsets.symmetric(horizontal: 12),
        ),
      ),
    );
  }
}
